docs/src/operate/lifecycle/deregister.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Deregister entities
+tags: [Lifecycle]
+sidebar_position: 1
+description: Deregister entities from thin-edge
+---
+
+# Deregister entities
+
+All the entities registered with thin-edge are stored with the MQTT broker as retained messages,
+with their metadata spread across multiple topics.
+For e.g, for a child device `child01`, the registration message is stored on the `te/device/child01//` topic,
+its twin data is stored across several `te/device/child01///twin/<twin-data-type>` topics,
+its command metadata stored across several `te/device/child01///cmd/<cmd-type>` topics and so on.
+
+On top of that, when a device has services associated with it, or has nested child devices,
+they all have their own respective registration topics and other metadata topics.
+So, de-registering a device involves de-registering itself, its metadata and
+the complete entity hierarchy that is associated with it.
+
+## Deregister a single entity
+
+De-registering a single entity, excluding its child entities, involves clearing its registration message and
+all the metadata messages associated with it from the MQTT broker.
+Even though thin-edge doesn't provide a direct API for this yet, this can be easily be done using third-party tools
+like `mosquitto_sub` that supports clearing multiple retained messages together using the `--remove-retained` option.
+
+For e.g, a child device with topic id `device/child01//` can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/child01///#"
+```
+
+The topic filter `te/device/child01///#` covers the device's registration message as well all of its metadata messages.
+Running the same command without the `remove-retained` option shows the list of messages that would be cleared with this command.
+The `-W 3` option is used to stop the command after a maximum timeout period of 3 seconds.
+Adjust this timeout value based on the number of entities in your deployment.
+
+:::note
+Deregistering the main device using this mechanism is not supported.
+:::
+
+Similarly, a service with topic id `device/main/service/service01` can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/main/service/service01/#"
```

Comment:
I'm fairly sure we can remove the timeout option, `-W 3` and replace it with `--retained-only` (but double check this, see [docs](https://mosquitto.org/man/mosquitto_sub-1.html)).

```suggestion
mosquitto_sub --remove-retained --retained-only -t "te/device/main/service/service01/#"
```

Reply:
Even though the doc of `--retained-only` states that "Remove all retained messages on the server, assuming we have access to do so, **and then exit**", I don't see that happening in practice, even after explicitly sending a non-retained message. Anyway, it would only work when there is some non-retained message on the given topic filter, which isn't always guaranteed, right?